pkg/northbound/gnmi/v2/subscribe.go

```diff
@@ -1,14 +1,182 @@
+// SPDX-FileCopyrightText: 2020-present Intel Corporation
 // SPDX-FileCopyrightText: 2020-present Open Networking Foundation <info@opennetworking.org>
 //
 // SPDX-License-Identifier: Apache-2.0
 
 package gnmi
 
 import (
+	"context"
+	"github.com/golang/protobuf/proto"
+	"github.com/grpc-ecosystem/go-grpc-middleware/util/metautils"
+	"github.com/onosproject/onos-api/go/onos/topo"
+	"github.com/onosproject/onos-lib-go/pkg/errors"
+	baseClient "github.com/openconfig/gnmi/client"
 	"github.com/openconfig/gnmi/proto/gnmi"
+	"io"
+	"strings"
 )
 
+type subContext struct {
+	ctx       context.Context
```

Comment:
we should not store context in struct. However it should be passed from one function to another and should flow through your program

FYI https://go.dev/blog/context-and-structs

Reply:
Done.